.github/workflows/release-compile-changelog.yml

```
@@ -62,11 +45,75 @@ jobs:
+              date.getFullYear() !== year ||
+              date.getMonth() !== month - 1 ||
+              date.getDate() !== day
+            ) {
+              core.setFailed('Invalid release date. The release date must be an actual date.');
+              process.exit(1);
+            }
+
+            console.log(`Release date validated: ${releaseDate}`);
+
       - name: Checkout code
         uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)